### PR TITLE
Add a few abbreviations and terms to the glossary

### PR DESCRIPTION
**Pull request by roelof-groenewald**

### CODE REVIEW

File: Docs/source/glossary.rst [hunk 1]
```diff
@@ -22,12 +22,14 @@ Abbreviations
 * **CI:** continuous integration, automated tests that we perform before a proposed code-change is accepted; see PR
 * **CPU:** `central processing unit <https://en.wikipedia.org/wiki/Central_processing_unit>`__, we usual mean a socket or generally the host-side of a computer (compared to the accelerator, e.g. GPU)
 * **DOE:** `The United States Department of Energy <https://en.wikipedia.org/wiki/United_States_Department_of_Energy>`__, the largest sponsor of national laboratory research in the United States of America
+* **DSMC:** Direct Simulation Monte Carlo, a method to capture collisions between kinetic particles
```
Comment (lucafedeli88) [inline Docs/source/glossary.rst]:
```suggestion
* **DSMC:** `Direct Simulation Monte Carlo <https://en.wikipedia.org/wiki/Direct_simulation_Monte_Carlo>`__, a method to capture collisions between kinetic particles
```
Should we add a link to the corresponding wikipedia page?

File: Docs/source/glossary.rst [hunk 2]
```diff
@@ -22,12 +22,14 @@ Abbreviations
...
 * **ECP:** `Exascale Computing Project <https://www.exascaleproject.org>`__, a :ref:`U.S. DOE funding source <funding>` that supports WarpX development
 * **ECT:** `Enlarged Cell Technique <https://ieeexplore.ieee.org/document/4463918>`__, an electromagnetic solver with accurate resolution of perfectly conducting embedded boundaries
 * **EB:** embedded boundary, boundary conditions inside the simulation box, e.g. following material surfaces
 * **EM:** electromagnetic, e.g. EM PIC
 * **ES:** electrostatic, e.g. ES PIC
 * **FDTD:** `Finite-difference time-domain or Yee's method <https://en.wikipedia.org/wiki/Finite-difference_time-domain_method>`__, a class of grid-based finite-difference field solvers
+* **FRC:** Field Reversed Configuration, an approach of magnetic confinement fusion
```
Comment (lucafedeli88) [inline Docs/source/glossary.rst]:
```suggestion
* **FRC:** `Field Reversed Configuration <https://en.wikipedia.org/wiki/Field-reversed_configuration>`__, an approach of magnetic confinement fusion
```

same as above

**Review by lucafedeli88 — APPROVED** [review]
Thanks a lot for this PR! I left two optional suggestions.

File: Docs/source/glossary.rst (hunk 1 above)
Comment (roelof-groenewald) [reply Docs/source/glossary.rst]: Sounds good to me, thanks for the suggestion!

**Pull request merged**